build-support/src/main/kotlin/aws/sdk/kotlin/gradle/crt/NativeUtils.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.sdk.kotlin.gradle.crt
+
+import org.gradle.api.Project
+import org.gradle.api.publish.maven.tasks.AbstractPublishToMaven
+import org.gradle.api.publish.tasks.GenerateModuleMetadata
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.withType
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
+import org.jetbrains.kotlin.konan.target.Family
+import org.jetbrains.kotlin.konan.target.HostManager
+import org.jetbrains.kotlin.konan.target.KonanTarget
+
+/**
+ * Kotlin/Native Linux and Windows targets are generally enabled on all hosts since
+ * the Kotlin toolchain and backend compilers support cross compilation. We
+ * are using cinterop and have to compile CRT for those platforms which sometimes
+ * requires using docker which isn't always available in CI or setup in users environment.
+ *
+ * See [KT-30498](https://youtrack.jetbrains.com/issue/KT-30498)
+ */
+fun Project.disableCrossCompileTargets() {
+    plugins.withId("org.jetbrains.kotlin.multiplatform") {
+        configure<KotlinMultiplatformExtension> {
+            targets.withType<KotlinNativeTarget> {
+                val knTarget = this
+                when {
+                    HostManager.hostIsMac && (knTarget.isLinux || knTarget.isWindows) -> disable(knTarget)
+                    HostManager.hostIsLinux && knTarget.isApple -> disable(knTarget)
+                    HostManager.hostIsMingw && (knTarget.isLinux || knTarget.isApple) -> disable(knTarget)
+                }
```

Comment:
**Question**: When `HostManager.hostIsLinux` we only disable if the target is Apple but not if it's Windows?

Reply:
Well we haven't enabled mingw as a target yet so I just missed it but also we should be able to cross compile mingw on linux. We'll have to see how this works out but this is _mostly_ a CI concern